crates/agent/tests/e2e_proposal_one.rs

```diff
@@ -0,0 +1,817 @@
+use agent::services::agent::AgentService;
+use anyhow::Result;
+use std::time::Duration;
+
+mod fixtures {
+    pub mod create_agent;
+    pub mod proposal_one_answers;
+    pub mod proposals;
+    pub mod proposals_questions;
+
+    pub use proposals::get_proposals;
+    // Use only one set of struct definitions to avoid duplicates
+    pub use create_agent::{create_agent_service, direct_query_agent};
+    pub use proposal_one_answers::get_proposal_one_answers;
+    pub use proposals_questions::get_proposal_questions;
+}
+
+use fixtures::{create_agent_service, direct_query_agent};
+
+/// Ask a question about a proposal to the agent and get the response
+/// Will retry up to 2 times if the response is empty
+async fn ask_agent_question(
+    agent_service: &AgentService,
+    proposal_text: &str,
+    question: &str,
+) -> Result<String> {
+    const MAX_RETRIES: usize = 2;
+    let mut retries = 0;
+    let mut last_answer = String::new();
+
+    while retries <= MAX_RETRIES {
+        // Create a prompt with both the proposal text and the question
+        let is_binary = question.to_lowercase().contains("(yes/no)");
+
+        let prompt = if is_binary {
+            if retries == 0 {
+                format!("PROPOSAL TEXT:\n\n{}\n\nQUESTION: {}\n\nPlease answer with ONLY 'yes' or 'no' followed by a brief explanation. Your answer MUST start with either 'yes' or 'no'.", proposal_text, question)
+            } else {
+                format!("PROPOSAL TEXT:\n\n{}\n\nQUESTION: {}\n\nCRITICAL INSTRUCTION: You MUST answer with ONLY 'yes' or 'no' followed by a brief explanation. Your answer MUST start with either 'yes' or 'no'. DO NOT provide any other response format.", proposal_text, question)
+            }
+        } else {
+            // For reasoning questions
+            if retries == 0 {
+                format!("PROPOSAL TEXT:\n\n{}\n\nQUESTION: {}\n\nPlease provide a direct and specific answer to the question based solely on the proposal text. Be concise but thorough.", proposal_text, question)
+            } else {
+                format!("PROPOSAL TEXT:\n\n{}\n\nQUESTION: {}\n\nCRITICAL INSTRUCTION: You MUST provide a direct and specific answer to the question based solely on the proposal text. Focus only on answering the question with factual information from the proposal.", proposal_text, question)
+            }
+        };
+
+        // Use our new direct_query_agent function for better responses
+        let answer = match direct_query_agent(agent_service, &prompt).await {
+            Ok(response) => response,
+            Err(e) => {
+                if retries < MAX_RETRIES {
+                    println!("Error on attempt {}: {:?}. Retrying...", retries + 1, e);
+                    retries += 1;
+                    tokio::time::sleep(Duration::from_millis(2000)).await; // Wait a bit before retrying
```

Comment:
We use different sleep amount between retries

Reply:
yeah make sense I will use 2 seconds only